extension/src/search.rs

```diff
@@ -181,16 +196,25 @@ pub fn search(
 
     let embeddings = transform(query, &project_meta.transformer, proj_api_key);
 
-    match project_meta.search_alg {
-        types::SimilarityAlg::pgv_cosine_similarity => cosine_similarity_search(
+    let result = match project_meta.index_dist_type {
+        types::IndexDist::pgv_hnsw_l2 => error!("Not implemented."),
+        types::IndexDist::pgv_hnsw_ip => error!("Not implemented."),
+        types::IndexDist::pgv_hnsw_cosine => cosine_similarity_search(
             job_name,
             &proj_params,
             &return_columns,
             num_results,
             &embeddings[0],
             where_clause,
         ),
+    };
+
+    // Added error handling for the result of the match
+    if let Err(err) = result {
+        return Err(err);
     }
+
+    Ok(result.unwrap())
```

Comment:
This does not seem like it is handling the error, rather it is unwrapping the result if its `Ok` then re-wrapping it in an Ok...

```suggestion
    match project_meta.index_dist_type {
        types::IndexDist::pgv_hnsw_l2 => error!("Not implemented."),
        types::IndexDist::pgv_hnsw_ip => error!("Not implemented."),
        types::IndexDist::pgv_hnsw_cosine => cosine_similarity_search(
            job_name,
            &proj_params,
            &return_columns,
            num_results,
            &embeddings[0],
            where_clause,
        ),
    }
```